src/main/java/org/springframework/social/openidconnect/HttpClientFactory.java

```diff
@@ -114,4 +116,26 @@ private static X509HostnameVerifier getVerifier(boolean isStrict){
         return SSLSocketFactory.STRICT_HOSTNAME_VERIFIER;
     }
 
+    /**
+     * Gets the TLS version from jdk parameter.  Defaults to version 1.2 for JDK 1.7 if not provided
+     * @return
+     */
+    public static String getTlsVersion() {
+        // don't use "jdk.tls.client.protocols" as it was causing unexpected exception during unit testing
+        String tlsVer = System.getProperty("tlsProtocol");
+
+        if(StringUtils.isEmpty(tlsVer)) {
+            double javaVersion = Double.parseDouble(System.getProperty("java.specification.version"));
+            if(javaVersion == 1.7) {
```

Comment:
I specified the concerns in the code comments.  JDK 1.6 throws exception when specifying tls 1.2, and JDK 1.8 already defaults to tls 1.2.  If we always default to tls 1.2 then we're definitely breaking older java versions.  

Even if default tls version isn't updated in future JDK versions for a some time, I don't think it's likely that they'll go back to an older tls version, but it's highly likely that tls 1.3 will become default eventually.

I just don't see why we would want to knowingly break older jdks and limit future jdks by default, since it's easy to avoid it.